src/auth/user-import-builder.ts

```diff
@@ -432,5 +480,6 @@ export class UserImportBuilder {
       }
       index++;
     });
+    return populatedUsers;
```

Comment:
If you wanted, you can just return a single array from this method containing both users and errors:

```
const populatedUsers = [];
users.forEach((user, index) => {
  try {
    const result = populateUploadAccountUser(user, userValidator);
    populatedUsers.push(result);
  } catch (error) {
    populatedUsers.push({index, error});
  }
})
return populatedUsers;
```

Then in the caller (constructor) you can make all the state changes:

```
const results = populateUsers(...);
this.validatedUsers = // filter out the users from results
this.indexMap = // construct the index map
this.requireHashOptions = // check if at least one user requires hash
```

I don't feel strongly about this, so I'll leave it to you.

Reply:
It is not easy constructing the indexMap after this completes. I have to remember the map original locations of the users to their new locations in the request sent to backend.